mindsdb/integrations/handlers/elasticsearch_handler/README.md

```diff
@@ -1,118 +1,179 @@
-PARAMETERS = {
-   "hosts": "127.0.0.1:9200",
-   "user": "user",
-   "password": "password"
-};
-```
+* `hosts`: The Elasticsearch host(s) in format "host:port". For multiple hosts, use comma separation like "host1:port1,host2:port2".
 
-Required connection parameters include the following (at least one of these parameters should be provided):
+Optional connection parameters include the following:
 
-* `hosts`: The IP address and port where ElasticSearch is deployed.
-* `user`: The user used to autheticate access.
-* `password`: The password used to autheticate access.
-</Tip>
+* `user`: The username for Elasticsearch authentication.
+* `password`: The password for Elasticsearch authentication.
+* `api_key`: API key for authentication (alternative to user/password).
+* `cloud_id`: Elastic Cloud deployment ID for hosted Elasticsearch.
+* `ca_certs`: Path to CA certificate file for SSL verification.
+* `client_cert`: Path to client certificate file for SSL authentication.
+* `client_key`: Path to client private key file for SSL authentication.
+* `verify_certs`: Boolean to enable/disable SSL certificate verification (default: true).
+* `timeout`: Request timeout in seconds.
 
 ## Usage
 
+The following usage examples utilize the connection to Elasticsearch made via the `CREATE DATABASE` statement and named `elasticsearch_conn`.
+
 Retrieve data from a specified index by providing the integration name and index name:
 
 ```sql
 SELECT *
-FROM elasticsearch_datasource.my_index
+FROM elasticsearch_conn.products
 LIMIT 10;
 ```
 
-<Note>
-The above examples utilize `elasticsearch_datasource` as the datasource name, which is defined in the `CREATE DATABASE` command.
-</Note>
+Query with filtering and aggregation:
+
+```sql
+SELECT category, COUNT(*) as product_count, AVG(price) as avg_price
+FROM elasticsearch_conn.products 
+WHERE price > 100
+GROUP BY category
+ORDER BY product_count DESC;
+```
+
+Run queries with array fields (automatically converted to JSON strings):
+
+```sql
+SELECT product_name, tags, categories 
+FROM elasticsearch_conn.products 
+WHERE product_id = '12345';
+```
 
 <Tip>
-At the moment, the Elasticsearch SQL API has certain limitations that have an impact on the queries that can be issued via MindsDB. The most notable of these limitations are listed below:
-1. Only `SELECT` queries are supported at the moment.
-2. Array fields are not supported.
-3. Nested fields cannot be queried directly. However, they can be accessed using the `.` operator.
+**Array Field Support**
 
-For a detailed guide on the limitations of the Elasticsearch SQL API, refer to the [official documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-limitations.html).
+The Elasticsearch handler automatically detects and converts array fields to JSON strings for SQL compatibility. This prevents "Arrays not supported" errors while preserving the original data structure.
 </Tip>
 
-## Troubleshooting Guide
+## Schema Discovery
```

Comment:
@richardokonicha I think this and the **Data Catalog Support** section need not be included in the documentation. The data catalog is primary used for Minds and it is not something that users will interact with much, so I don't think it is necessary.

If they want to take a look at their tables, they can always do `SHOW TABLES IN <elasticsearch-database>`.